src/main/java/touro/chess/KingPiece.java

```diff
@@ -0,0 +1,40 @@
+package touro.chess;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class KingPiece extends AbstractPiece{
+    public KingPiece(Location location, PieceColor color) {
+        super(location, color);
+    }
+
+    @Override
+    public List<Move> getMoves() {
+        Location currentLocation = getLocation();
+        int currentRow = currentLocation.getRow();
+        int currentColumn = currentLocation.getColumn();
+
+        //all possible new locations
+        List<Location> possibleLocations = Arrays.asList(
+                new Location(currentRow + 1, currentColumn),
+                new Location(currentRow - 1, currentColumn),
+                new Location(currentRow,currentColumn + 1),
+                new Location(currentRow, currentColumn - 1),
+                new Location(currentRow + 1, currentColumn -1),
+                new Location(currentRow -1, currentColumn -1),
+                new Location(currentRow + 1, currentColumn + 1),
+                new Location(currentRow -1, currentColumn + 1)
+        );
+        //check if the moves in possibleLocations are legal moves and then return list of all legal moves
+        ArrayList<Move> legalMoves = new ArrayList<>();
+        for (Location newLocation : possibleLocations){
+            Move newMove = new Move(currentLocation,newLocation,false);
+            Board board = new Board();
```

Comment:
This would be creating a new empty board instead of checking on the board of the current game